tests/models/telechat2/test_modeling_telechat2.py

```diff
@@ -0,0 +1,601 @@
+# coding=utf-8
+# Copyright 2025 The TeleChat team and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Testing suite for the PyTorch TeleChat2 model."""
+
+import gc
+import unittest
+
+import pytest
+from packaging import version
+from parameterized import parameterized
+
+from transformers import AutoTokenizer, TeleChat2Config, is_torch_available, set_seed
+from transformers.generation.configuration_utils import GenerationConfig
+from transformers.testing_utils import (
+    backend_empty_cache,
+    require_bitsandbytes,
+    require_flash_attn,
+    require_torch,
+    require_torch_sdpa,
+    slow,
+    torch_device,
+)
+
+from ...models.llama.test_modeling_llama import LlamaModelTest, LlamaModelTester
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import ids_tensor
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import (
+        TeleChat2ForCausalLM,
+        TeleChat2ForQuestionAnswering,
+        TeleChat2ForSequenceClassification,
+        TeleChat2ForTokenClassification,
+        TeleChat2Model,
+    )
+
+
+class TeleChat2ModelTester(LlamaModelTester):
```

Comment:
Here let's do like `test_modeling_gemma2` where we just specify the class and inherit the rest